### ISSUE TITLE
"editor.parameterHints.enabled": false does not work (plz help me)

### ISSUE THREAD

**Issue by asdoo4069** [microsoft/vscode-cpptools#9314]
Issue Type: <b>Bug</b>

Hi, I'm korean. So plz understand to my terrible English.
And I'm a noob :(
This is my first post at Github.

### I added `"editor.parameterHints.enabled": false` that setting in my `settings.json` file. But it does not work.

I usually work at C language.
While I type C source code (*.c)
( example `printf()`, `scanf()` )

when I type a something message in bracket(like `printf();`, `scanf();`), I saw `int printf(const char *const _Format, ...)` in that parameter hint.

After I changed `"editor.parameterHints.enabled": false` this,
`int printf(const char *const _Format, ...)` this hint is not appear when I type `printf()` manually. 
But when I use autocomplete(intellisense C/C++ Extension),
that parameter hint apper every each time!!! 

In conclusion, `"editor.parameterHints.enabled": false`
I type manually(one spell by one spell), that parameter hint is invisible.
I type with autocomplete, that parameter hint appear.

It's really annoying me.
plz comment. Thanks.

Extension version: 1.9.8
VS Code version: Code 1.67.1 (da15b6fd3ef856477bf6f4fb29ba1b7af717770d, 2022-05-06T12:37:03.389Z)
OS version: Windows_NT x64 10.0.19044
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-7820HQ CPU @ 2.90GHz (8 x 2904)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|15.88GB (8.85GB free)|
|Process Argv|--crash-reporter-id 864db12d-035b-4e94-aeed-12f680be1154|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
vswsl492cf:30256860
pythonvspyl392:30443607
pythontb:30283811
pythonvspyt551cf:30345471
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
vscoreces:30445986
pythondataviewer:30285071
vscod805:30301674
pythonvspyt200:30340761
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
vsaa593:30376534
vsc1dst:30438360
pythonvs932:30410667
wslgetstarted:30449410
pythonvsnew555:30457759
vscscmwlcmt:30465135
cppdebugcf:30475217
vsclangdc:30486549

```

</details>

**Comment by asdoo4069** [comment]
I think that problem is related to `"C_Cpp.autocompleteAddParentheses": true`.
I changed that setting to false, and then `"editor.parameterHints.enabled": false` is work!
But I don't know why those two settings are crashed. Is that bug??
If I remove VSCode completely and reinstall, it could be better?

**Comment by asdoo4069** [comment]
I want to use both `"C_Cpp.autocompleteAddParentheses": true` and `"editor.parameterHints.enabled": false`.
How can I use those two settings together?

**Comment by sean-mcmanus** [comment]
Yeah, our autocompleteAddParentheses feature triggers the `editor.action.triggerParameterHints` command -- we could check that setting before triggering the parameter hint.

**Comment by asdoo4069** [comment]
Sorry, I couldn't understand very well. Because I'm not good at English.

I want to use `C_Cpp.autocompleteAddParentheses` and also I don't want to use `editor.parameterHints.enabled`.
1. Can't I use those two settings at the same time?

2. What is the function of the code? `editor.action.triggerParameterHints`

**Comment by sean-mcmanus** [comment]
It's a bug or missing feature behavior -- we don't follow the editor.parameterHints.enabled setting when we call the "editor.action.triggerParameterHints" command, and it looks like VS Code itself doesn't change the behavior itself when that setting is called.

**Comment by sean-mcmanus** [comment]
The fix is available with 1.13.3 (pre-release): https://github.com/microsoft/vscode-cpptools/releases/tag/v1.13.3
